packages/babel-plugin-proposal-object-rest-spread/src/index.js

```diff
@@ -240,6 +240,27 @@ export default declare((api, opts) => {
     }
   }
 
+  function hasMoreThanOneBinding(node) {
+    if (t.isArrayPattern(node)) {
+      const nonNullElements = node.elements.filter(element => element !== null);
+      if (nonNullElements.length > 1) return true;
+      else return hasMoreThanOneBinding(nonNullElements[0]);
+    } else if (t.isObjectPattern(node)) {
+      if (node.properties.length > 1) return true;
+      else if (node.properties.length === 0) return false;
+      else return hasMoreThanOneBinding(node.properties[0]);
+    } else if (t.isObjectProperty(node)) {
+      return hasMoreThanOneBinding(node.value);
+    } else if (t.isAssignmentPattern(node)) {
+      return hasMoreThanOneBinding(node.left);
+    } else if (t.isRestElement(node)) {
+      return hasMoreThanOneBinding(node.argument);
+    } else {
+      // node is Identifier or MemberExpression
+      return true;
```

Comment:
Q: Since this is a single identifier, shouldn't it return false? (it's one binding)

Reply:
It looks like this test case fails if I make this false

input:
```
const { x15: [...{ ...y15 }] } = z();
```

output:
```
const {} = z(),
      y15 = babelHelpers.extends({}, z().x15);
```

Reply:
Ah I believe I have a fix. In the event  `node.argument` is an Identifier then we can early return true and this these test cases work out.

Reply:
Actually, I think we don't need to return `true` for rest but the fix needs to live somewhere else.

In the `const { x15: [...{ ...y15 }] } = z();` case, we don't need to cache `z()`: we can just skip the `{} = z()` one.

Reply:
Would it be ok to consider making this change in a separate PR as an optimization? My main goal for this is to mitigate duplicating the right hand assignment. I'd prefer avoiding doing a larger refactor of the greater plugin if possible.

Reply:
Yes, that's ok!